arch/xtensa/include/xtensa-asm2-s.h

```diff
@@ -356,6 +356,11 @@ _restore_\@:
  * with a simple jump instruction.
  */
 .macro DEF_EXCINT LVL, ENTRY_SYM, C_HANDLER_SYM
+#if defined(CONFIG_IMX) && (MEM_VECT_TEXT_SIZE <= 0x1C)
+.pushsection .iram.text, "ax"
+.global _Level\LVL\()VectorHelper
+_Level\LVL\()VectorHelper:
```

Comment:
This seems a little suspicious.  What's the problem being solved here?  Maybe you don't have enough space in the vector table?  FWIW: there are some thought about how to better generate the entry points that might help with that.

Reply:
For IMX the size constraint of the interrupt vector table entry is 0x1C bytes of code, less than usual.
I've added this indirection to bypass this size constraint and moved the default handlers to the end of vector table,  renaming them to _Level\LVL\()VectorHelper.

For the entry point I've kept the exact one we have in SOF with XTOS - _ResetVector in XTOS or _start (for Zephyr) - see https://github.com/thesofproject/sof/blob/main/src/platform/imx8/imx8.x.in#L136